build_debian.sh

```diff
@@ -618,6 +619,56 @@ then
 
 fi
 
+# #################
+#   secure boot 
+# #################
+if [[ $SECURE_UPGRADE_MODE == 'dev' || $SECURE_UPGRADE_MODE == "prod" && $SONIC_ENABLE_SECUREBOOT_SIGNATURE != 'y' ]]; then
+    # note: SONIC_ENABLE_SECUREBOOT_SIGNATURE is a feature that signing just kernel, 
+    # SECURE_UPGRADE_MODE is signing all the boot component including kernel.
+    # its required to do not enable both features together to avoid conflicts.
+    echo "Secure Boot support build stage: Starting .."
+
+    # debian secure boot dependecies
+    sudo LANG=C DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT apt-get -y install      \
```

Comment:
Are we downloading prebuilt binaries of SHIM and GRUB?

Reply:
yes, we are downloading the shim and grub from Debian sources unsigned, them the build sign them and copy to the corresponding boot location.

Reply:
Isn't this security gap? Who does auditing on these prebuilt images ? If SONiC vendor is signing these images then the SONiC vendor should able to audit them right or why not take Debian distribution signed images directly.

Reply:
well, SONiC downloads many Debian packages from Debian sources, so probably should be required to do a robust solution, because even if you audit the download of secure boot loaders components still SONiC it's downloading some kernel modules or some tools from some Debian sources without audit.
In general, the solution should be to have some server as a repository to contain all the deb resources and authenticate them...
About the other question, in our HLD review, we are giving the Vendor the ability to sign with his own keys so they are not limitate to using the Microsoft pub key only.

Reply:
i am not sure sonic is downloading any kernel binary modules, all kernel and modules are build from sources.

Reply:
in the build_debian.sh many pkgs are downloading and not build from source if you search for the word "apt-get" you will find, i.e like this one:
`sudo LANG=C chroot $FILESYSTEM_ROOT apt-get -y install apt-transport-https \
                                                       ca-certificates \
                                                       curl \
                                                       gnupg2 \
                                                       software-properties-common

`
But it's not a big change for me, so if its better I can download it in a Makefile, similar like we did with src/iptables.

Reply:
The source code unfortunately has also a vulnerability...
Well, I found that many packages are getting the sources code from http debian sources instead https. I already raise this issue to the community.
This issue can affect many features not just the SHIM and GRUB.

But, like I mentioned before, from my side it's not a big modification and I will do this small modification to GRUB & SHIM.
I probably will do a similar solution creating a Makefile like was used with iptables sources, but I will use a https source instead http source.
My plan is to use this source: 
https://deb.debian.org/debian/pool/main/g/grub2/

pls let me know if you have still any concern, thanks

Reply:
Yes, if you could bring both SHIM and GRUB like iptable src would be good. Cisco can also add this as separate PR. We have raised PR. https://github.com/sonic-net/sonic-buildimage/pull/12901, OR you can also bring it that.

Reply:
We can modify that PR so that it can go on top of this PR.

Reply:
Let us know.

Reply:
I'd suggest that we commit this PR with our current changes and then as an improvement to the PR - you'll commit your changes with the adjusted shim & grub src in a follow up PR

Reply:
@Yarden-Z - the current PR has this gap/comment that needs to be addressed.  Could this PR be updated/separated so that it does not download pre-built binaries?